src/main/java/io/vertx/ext/amqp/impl/AmqpConnectionImpl.java

```diff
@@ -268,23 +271,62 @@ public AmqpConnection createReceiver(String address, AmqpReceiverOptions receive
     runWithTrampoline(v -> {
       ProtonReceiver receiver = connection.get().createReceiver(address, opts);
 
-      if (receiverOptions != null && receiverOptions.getQos() != null) {
-        receiver.setQoS(ProtonQoS.valueOf(receiverOptions.getQos().toUpperCase()));
+      if (receiverOptions != null) {
+        if (receiverOptions.getQos() != null) {
+          receiver.setQoS(ProtonQoS.valueOf(receiverOptions.getQos().toUpperCase()));
+        }
+
+        List<String> desired = receiverOptions.getDesiredCapabilities();
+        List<String> provided = receiverOptions.getCapabilities();
+
+        receiver.setDesiredCapabilities(desired.stream().map(Symbol::valueOf).toArray(Symbol[]::new));
+        receiver.setOfferedCapabilities(provided.stream().map(Symbol::valueOf).toArray(Symbol[]::new));
+
+        configureTheSource(receiverOptions, receiver);
       }
 
-      runWithTrampoline(x -> {
-        new AmqpReceiverImpl(address, this, receiver, handler, completionHandler);
-      });
+      new AmqpReceiverImpl(address, this, receiverOptions != null && receiverOptions.isDurable(),
+        receiver, handler, completionHandler);
     });
     return this;
   }
 
+  private void configureTheSource(AmqpReceiverOptions receiverOptions, ProtonReceiver receiver) {
+    org.apache.qpid.proton.amqp.messaging.Source source = (org.apache.qpid.proton.amqp.messaging.Source) receiver
+      .getSource();
+    if (receiverOptions.isDurable()) {
+      source.setExpiryPolicy(TerminusExpiryPolicy.NEVER);
+      source.setDurable(TerminusDurability.UNSETTLED_STATE);
+    } else {
+      // Check if we have individual values, not the in this case the receiver won't be considered as durable and the
```

Comment:
I wouldn't have this disparity in its current state. It seems unwise to allow controlling the terminus durability manually without also offering some way for requesting a detach with closed=false, since the main thing folks would want to use this for is the same, and so setting closed=true will then typically fail to achieve what they wanted. I'd suggest either get rid of the 'durable' shortcut, or make this detect the same case and act the same, or add an option to control what happens when close() is called (or add a detach method).

Reply:
..or get rid of ability to manually set the details, and just leave the durable option shortcut for now.